labrad/protocol.py

```diff
@@ -416,20 +415,34 @@ def require_secure_connection(auth_type):
 
         if isinstance(credential, auth.Password):
             if not credential.username:
-                # send login packet to get password challenge
-                challenge = yield self._sendManagerRequest()
-
-                # send password response
-                m = hashlib.md5()
-                m.update(challenge)
-                if isinstance(credential.password, bytes):
-                    m.update(credential.password)
+                if 'srp' in self.manager_features:
+                    # send identity
+                    group, salt, B_bytes = yield self._sendManagerRequest(10, username)
+
+                    # send password response
+                    srp = crypto.SRP(identity=username, group=group,
+                                     password=credential.password)
+                    A_bytes, M1 = srp.process_server_challenge(salt, B_bytes)
+                    try:
+                        resp, M2 = yield self._sendManagerRequest(11, (A_bytes, M1))
+                    except Exception:
```

Comment:
I realize that labrad will propagate the stack trace to the client, which should hopefully explain where the exception is being raised, but I'm still hesitant to use this outside of generic code. Any chance we can be more specific in this exception? My experience with `except Exception` is that it can cause serious debugging issues trying to isolate root causes.